src/SipHash.chpl

```diff
@@ -151,23 +181,41 @@ module SipHash {
         if DEBUG {
             try! {
               shLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-                                                   "%i v0 %016xu".format(D.size, v0));
+                                                   "%i v0 %016xu".format(numBytes, v0));
               shLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-                                                   "%i v1 %016xu".format(D.size, v1));
+                                                   "%i v1 %016xu".format(numBytes, v1));
               shLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-                                                   "%i v2 %016xu".format(D.size, v2));
+                                                   "%i v2 %016xu".format(numBytes, v2));
               shLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-                                                   "%i v3 %016xu".format(D.size, v3));
+                                                   "%i v3 %016xu".format(numBytes, v3));
             }
         }
     }
 
-    for pos in D.low..lastPos-1 by 8 {
-        if isSubtype(msg.type, c_ptr) {
-          m = U8TO64_LE(msg + pos);
+    for pos in D.low..lastPos-1 by stride {
+      // select (stride, isSubtype(msg.type, c_ptr))
+        if (stride == 8) {
+          if isSubtype(msg.type, c_ptr) {
+            m = U8TO64_LE(msg + pos);
+          } else {
+            m = U8TO64_LE(msg, pos..#stride);
+          }
+        } else if (stride == 1) {
+          if isSubtype(msg.type, c_ptr) {
+            m = XTO64_LE(msg + pos);
+          } else {
+            m = XTO64_LE(msg[pos]);
+          }
         } else {
-          m = U8TO64_LE(msg, pos..#8);
+          // Should never reach here
+          compilerError("can only hash (arrays of) 8-bit or 64-bit values");
```

Comment:
Note that this is a compile-time / build-time error (not a runtime error that is `thrown`)

Reply:
gotcha, okay

Reply:
This actually is not a thrown error, but a compile-time error. I agree it would be nice to get module/routine/line info in a compiler error, though. I'll play around with this.